base/abstractarray.jl

```diff
@@ -150,6 +150,7 @@ prevind(::AbstractArray, i::Integer) = Int(i)-1
 nextind(::AbstractArray, i::Integer) = Int(i)+1
 
 eltype(::Type{<:AbstractArray{E}}) where {E} = @isdefined(E) ? E : Any
+elsize(::Type{<:AbstractArray{T}}) where {T} = aligned_sizeof(T)
```

Comment:
This doesn't feel right to me, either.  The default `write(::AbstractArray)` just does:

```Julia
    for a in A
        nb += write(s, a)
    end
```

Which won't include alignment padding by default.

Reply:
I guess that's right --- it seems we can't fix this more thoroughly without more drastic or breaking changes. For now we should probably do your PR + define `elsize` for `SubArray{<:Any,<:Any,<:Array}` + just the `write` change from here.

In the future we should consider writing alignment padding in `write(::AbstractArray)`, and possibly exporting `elsize`.